### PR TITLE
VSP-2360: Only pass optional billing params when needed

### PR DESCRIPTION
**Pull request by Nicole8164**
Billing information is optional so we should only pass it when it is needed

### CODE REVIEW

File: src/Message/Astropay/AuthorizeRequest.php [hunk 1]
```diff
@@ -232,6 +232,37 @@ public function setBillingPostcode($value)
...
+    {
+        $this->setParameter('billing_info_enabled', $value);
+    }
+
+    /**
+     * Gets if the billing info should be passed in request
+     */
+    protected function getBillingInfoEnabled()
+    {
+        return $this->getParameter('billing_info_enabled');
```
Comment (ybenzine) [inline src/Message/Astropay/AuthorizeRequest.php]: 👏  👏  👏

File: tests/Message/Astropay/AuthorizeRequestTest.php [hunk 2]
```diff
@@ -64,14 +64,25 @@ public function testGetData()
...
-        $this->assertEquals(ksort($result), ksort($expected_results));
+        $this->assertEquals($expected_results, $result);
+    }
+
+    public function testGetDataSetsBillingInformationWhenEnabled()
+    {
+        $this->request->setBillingInfoEnabled(true);
+        $this->request->setBillingStreet1('Main St');
+        $this->request->setBillingStreet2('Apt 1');
+        $this->request->setBillingCity('Sao Paul');
```
Comment (ybenzine) [inline tests/Message/Astropay/AuthorizeRequestTest.php]: I know it's just a test and doesn't matter but you're missing the last 'o' in `São Paulo` 😄

File: tests/Message/Astropay/AuthorizeRequestTest.php [hunk 3]
```diff
@@ -64,14 +64,25 @@ public function testGetData()
...
+        $this->request->setBillingPostcode('01000');
+
+        $result = $this->request->getData();
+        $this->assertEquals($result['billing.street1'], 'Main St');
+        $this->assertEquals($result['billing.street2'], 'Apt 1');
+        $this->assertEquals($result['billing.city'], 'Sao Paul');
```
Comment (ybenzine) [inline tests/Message/Astropay/AuthorizeRequestTest.php]: and here... 🙂

File: src/Message/Astropay/AuthorizeRequest.php [hunk 4]
```diff
@@ -232,6 +232,37 @@ public function setBillingPostcode($value)
...
+     * @param bool $value
+     */
+    public function setBillingInfoEnabled($value)
+    {
+        $this->setParameter('billing_info_enabled', $value);
+    }
+
+    /**
+     * Gets if the billing info should be passed in request
+     */
```
Comment (joecollier) [inline src/Message/Astropay/AuthorizeRequest.php]: Should we denote that this returns a bool?

**Pull request merged**